app/next-client-app/components/scanreports/HandleArchive.tsx

```diff
@@ -0,0 +1,29 @@
+import { archiveDataSets } from "@/api/datasets";
+import { updateScanReport } from "@/api/scanreports";
+import { toast } from "sonner";
+
+export const HandleArchive = async ({
+  id,
+  hidden,
+  ObjName,
+  type,
+}: {
+  id: number;
+  hidden: boolean;
+  ObjName: string;
+  type: string;
+}) => {
+  const message = hidden ? "Unarchive" : "Archive";
+  let response;
+  if (type === "datasets") {
```

Comment:
We will likely add a third type of data to archive, so better to use a switch here.